### ISSUE TITLE
adding bar plot to `%sqlcmd plot`

### ISSUE THREAD

**Issue by edublancas** [ploomber/jupysql#417]
We should add `%sqlcmd plot bar` to create bar plots.

The SQL should be pretty simple:

```sql
select col, count(*)
from table
group by col
```

then, we take the col, count pairs and use them to create the plot.

cc @jorisroovers - tagging you so you're in the loop!

**Comment by edublancas** [comment]
I've never been a fan of pie charts 😂, humans aren't good at judging areas. I like normalized bar charts better - they can help to visualize the same data.

But I think @jorisroovers mention they use it. what do you think?

**Comment by edublancas** [comment]
ok so let's add `%sqlcmd plot bar` and `%sqlcmd plot pie`

@mehtamohit013: please write acceptance criteria.

**Comment by mehtamohit013** [comment]
AC Criteria:
- Add Bar plot and pie chart to ggplot API + Test cases
- Modify the documentation and user guide to include both
- Changelog

**Comment by edublancas** [comment]
> Add Bar plot and pie chart to ggplot API + Test cases

First ensure that the tests pass with duckdb, once that's done, we can evaluate which other DBs we'll support.

> Modify the documentation and user guide to include both

add the examples to the existing section in the [docs](https://jupysql.ploomber.io/en/latest/api/magic-plot.html)

btw, the commands should be in `%sqlplot` (not in `%sqlcmd plot`), I'll fix the references in my previous comments to avoid confusion.

**Comment by mehtamohit013** [comment]
@edublancas 
So, if I understood correctly:
There are two APIs for plotting: `ggplot` and `%sqlplot` and both internally rely on `src/sql/plot.py` to plot both histogram and boxplot. So the scope of the issue would be to add the `bar` and `pie` class to the `src/sql/plot.py` and integrate it would `%sqlplot`. Also, I have to check if there are any special functions required to calculate these when executing SQL queries and check which DBs support it.

**Comment by edublancas** [comment]
right. integrate it with `%sqlplot` and add your code to `plot.py`, later we can define if we also offer this as part of the ggplot API

> Also, I have to check if there are any special functions required to calculate these when executing SQL queries and check which DBs support it.

Yes, I'd say first get DuckDB working and once that's ready we can add a test to our integration tests to find out which ones are passing and which ones are failing, then we define what we do

**Comment by mehtamohit013** [comment]
> The SQL should be pretty simple:
> 
> ```sql
> select col, count(*)
> from table
> group by col
> ```
> 
> then, we take the col, count pairs and use them to create

I was thinking more generic, letting the user pass the two columns `x` and `height` as done in [matplotlib API](https://matplotlib.org/stable/api/_as_gen/matplotlib.pyplot.bar.html) and we simply plot it.

Also, is there a need for stacked bar graph (I don't know if that's a thing or not)?

**Comment by edublancas** [comment]
> I was thinking more generic, letting the user pass the two columns x and height as done in [matplotlib API](https://matplotlib.org/stable/api/_as_gen/matplotlib.pyplot.bar.html) and we simply plot it.

we want JupySQL to be a higher-level API than matplotlib. If you think from the perspective of a data analyst using JupySQL, they want to explore the data quickly to understand it, and computing the height is an extra step that we want to avoid. Users should only point us to a column and we should group and plot the bar/pie chart

**Comment by edublancas** [comment]
let's leave out stacked or multiple-column charts for now

**Comment by mehtamohit013** [comment]
> we want JupySQL to be a higher-level API than matplotlib. If you think from the perspective of a data analyst using JupySQL, they want to explore the data quickly to understand it, and computing the height is an extra step that we want to avoid. Users should only point us to a column and we should group and plot the bar/pie chart

@edublancas 
For the pie chart, should we do two columns one for label and another for size? or the same as the bar chart?

**Comment by edublancas** [comment]
> For the pie chart, should we do two columns one for label and another for size? or the same as the bar chart?

same as bar chart. the command should only take a column as an argument, and we should compute the percentages, then pass this to the plotting function
